piptools/resolver.py

```diff
@@ -87,14 +88,17 @@ def resolve(self, max_rounds=10):
 
     def _check_constraints(self):
         for constraint in chain(self.our_constraints, self.their_constraints):
-            if constraint.link is not None and not constraint.editable:
-                msg = ('pip-compile does not support URLs as packages, unless they are editable '
-                       '(perhaps add -e option?)')
-                raise UnsupportedConstraint(msg)
+            if constraint.link and constraint.link.scheme not in vcs.all_schemes:
```

Comment:
Is there perhaps a convenient property on a `Link` object in the `pip` project that would expose this is-this-a-vcs-url check (`constraint.link.scheme not in vcs.all_schemes`) publicly without having to rely on implementation details like fetching that `vcs` object? Ideally I don't want to know about this at all.